lib/icingadb/redisconnection.hpp

```diff
@@ -387,6 +394,67 @@ void RedisConnection::WriteOne(StreamPtr& stream, RedisConnection::Query& query,
 	}
 }
 
+/**
+ * Initialize a Redis stream
+ *
+ * @param stream Redis server connection
+ * @param query Redis query
+ */
+template<class StreamPtr>
+void RedisConnection::Handshake(StreamPtr& stream, boost::asio::yield_context& yc)
+{
+	auto strm (stream);
```

Comment:
To keep a reference.

Reply:
But you already have one, It's called `stream`. If that one might become invalid during the function call, copying it won't help.

Reply:
Of course copying will help. Once copied, I have a non-dangling ref. See also other such template functions in the same file.

Reply:
If you'd for example capture it in a lambda that lives longer than this function yes, but this doesn't happen here (and neither in the other functions in that file if I haven't missed something). And if that was an issue here, you don't have any guarantee that your copy happens before the original becomes invalid, so it would still be broken.